crates/hir_def/src/nameres/tests/macros.rs

```diff
@@ -837,3 +837,25 @@ fn collects_derive_helpers() {
         _ => unreachable!(),
     }
 }
+
+#[test]
+fn resolve_macro_def() {
+    check(
+        r#"
+//- /lib.rs
+pub macro structs($($i:ident),*) {
+    $(struct $i { field: u32 } )*
+}
+
+structs!(Foo);
+
+//- /nested.rs
+structs!(Bar, Baz);
+"#,
+        expect![[r#"
+            crate
+            Foo: t
+            structs: m
+        "#]],
+    );
+}
```

Comment:
Can you also add a test that something like the following doesn't resolve the item outside the macro?

```rust
//- /lib.rs
pub macro struct() {
    struct Foo { field: u32 }
}
struct!();
```

Reply:
We still don't have macro hygiene yet, so `Foo` will be valid in your case anyway...